### ISSUE TITLE
Allow choice of whether to double encode within ActiveForm

### ISSUE THREAD

**Issue by cyphix333** [yiisoft/yii2#12988]
I created a [forum thread](http://www.yiiframework.com/forum/index.php/topic/73074-stop-yii-double-encoding-html-entities-with-activeform/page__gopid__305050) asking whether it was possible to stop Yii double encoding data it automatically populates form fields with when you use ActiveForm, but from the response I got, it seems it's hardcoded to double-encode data it puts into the fields.

As pointed out by @softark in _yii\helpers\BaseHtml_:

```
    public static function encode($content, $doubleEncode = true)
    {
        return htmlspecialchars($content, ENT_QUOTES | ENT_SUBSTITUTE, Yii::$app ? Yii::$app->charset : 'UTF-8', $doubleEncode);
    }

    ...

    public static function textarea($name, $value = '', $options = [])
    {
        $options['name'] = $name;
        return static::tag('textarea', static::encode($value), $options);
    }
```
You will note that the line `static::encode($value)` does not allow the customization of the `$doubleEncode` parameter, this forcing the default value of `true`.

Is it possible to add a property to **ActiveForm** where you can set the default value and then override the default if needed for each specific field with a configuration option within **ActiveField**?

My use case as you can see in the forum post is that I wish to pass entities to the field without them being `double-encoded`.

### What steps will reproduce the problem?

Set a model property that is used with **ActiveForm** to content that contains html entities, of which that will be loaded into a form field automatically by Yii.

### What is the expected result?

To not double-encode entities.

### What do you get instead?

Existing entities get encoded again, thus causing incorrect display.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.9
| PHP version      | 5.5.28
| Operating system | Win 7 / CentOS

**Comment by SilverFire** [comment]
Yeah, I think it could be done via `$options` parameter of `ActiveField::textarea()` method.

Would you like to try to submit a PR?

**Comment by cyphix333** [comment]
Thanks for the response.

Isn't the `$options` parameter only supposed to be used for attribute name-value pairs though?

**Comment by SilverFire** [comment]
Yes it is, but there is a way to pass some special options. See [this](https://github.com/yiisoft/yii2/blob/master/framework/widgets/ActiveField.php#L75) as an example

**Comment by cyphix333** [comment]
Oh ok, I guess that could work then. :) I will see if I can get around to doing a pull request for it.

**Comment by SilverFire** [comment]
Thank you!

**Comment by cyphix333** [comment]
@SilverFire Sorry, but what does "Breaks BC" mean in the pull request?

Backwards Compatibility?

**Comment by samdark** [comment]
Yes.

**Comment by cebe** [comment]
fixed by #12994
